src/backend/distributed/planner/distributed_planner.c

```diff
@@ -147,24 +144,10 @@ distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 	}
 	else if (CitusHasBeenLoaded())
 	{
-		if (IsLocalReferenceTableJoin(parse, rangeTableList))
+		needsDistributedPlanning = ListContainsDistributedTableRTE(rangeTableList);
+		if (needsDistributedPlanning)
 		{
-			/*
-			 * For joins between reference tables and local tables, we replace
-			 * reference table names with shard tables names in the query, so
-			 * we can use the standard_planner for planning it locally.
-			 */
-			UpdateReferenceTablesWithShard(rangeTableList);
-
-			needsDistributedPlanning = false;
-		}
-		else
-		{
-			needsDistributedPlanning = ListContainsDistributedTableRTE(rangeTableList);
-			if (needsDistributedPlanning)
-			{
-				fastPathRouterQuery = FastPathRouterQuery(parse, &distributionKeyValue);
-			}
+			fastPathRouterQuery = FastPathRouterQuery(parse, &distributionKeyValue);
```

Comment:
not related to this PR, but for a general suggestion in our codebase I suggest using `is` or similar prefixes for booleans because `fastPathRouterQuery` is not explicit about the type, which might make reading a bit harder. It could mean `a query` or `a boolean to indicate that it is a query`. without looking at the type, we don't know which one it is. If it was `isFastPathRouterQuery` it could be easier to understand IMHO. Just writing as a thought